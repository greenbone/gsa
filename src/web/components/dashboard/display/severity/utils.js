/* SPDX-FileCopyrightText: 2025 Greenbone AG
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import FilterTerm from 'gmp/models/filter/filterterm';
import {isDefined} from 'gmp/utils/identity';
import {LOG_VALUE} from 'web/utils/severity';

/**
 * Converts filter values to filter terms based on severity range.
 *
 * @param {Object} params - The filter values.
 * @param {(string|number)} params.start - The start value of the filter range.
 * @param {(string|number)} params.end - The end value of the filter range.
 * @returns {FilterTerm[]} An array of FilterTerm objects representing the filter terms.
 */
export const filterValueToFilterTerms = ({start, end}) => {
  if (isDefined(end)) {
    const startTerm = FilterTerm.fromString(`severity>${start}`);
    const endTerm = FilterTerm.fromString(`severity<${end}`);
    return [startTerm, endTerm];
  }
  if (start > LOG_VALUE) {
    return [FilterTerm.fromString(`severity>${start}`)];
  }
  return [FilterTerm.fromString(`severity=${start}`)];
};
