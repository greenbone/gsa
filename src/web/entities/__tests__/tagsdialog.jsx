/* SPDX-FileCopyrightText: 2024 Greenbone AG
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import {describe, test, expect, testing} from '@gsa/testing';
import {getDialog, getSelectElement} from 'web/components/testing';
import {render, fireEvent} from 'web/utils/testing';

import Dialog from '../tagsdialog';

describe('TagsDialog dialog component tests', () => {
  test('should render dialog', () => {
    const handleClose = testing.fn();
    const handleNewTagClick = testing.fn();
    const handleSave = testing.fn();
    const handleTagChange = testing.fn();

    render(
      <Dialog
        entitiesCount={3}
        tags={[{name: 'foo', id: '123'}]}
        onClose={handleClose}
        onNewTagClick={handleNewTagClick}
        onSave={handleSave}
        onTagChanged={handleTagChange}
      />,
    );

    expect(getDialog()).toBeInTheDocument();
  });

  test('should disable tag selection when no options available', () => {
    const handleClose = testing.fn();
    const handleNewTagClick = testing.fn();
    const handleSave = testing.fn();
    const handleTagChange = testing.fn();

    render(
      <Dialog
        entitiesCount={3}
        onClose={handleClose}
        onNewTagClick={handleNewTagClick}
        onSave={handleSave}
        onTagChanged={handleTagChange}
      />,
    );

    expect(getSelectElement()).toBeDisabled();
  });

  test('should save data', () => {
    const handleClose = testing.fn();
    const handleNewTagClick = testing.fn();
    const handleSave = testing.fn();
    const handleTagChange = testing.fn();

    const {getByTestId} = render(
      <Dialog
        comment="foo"
        entitiesCount={3}
        name="bar"
        tagId="123"
        tags={[{name: 'foo', id: '123'}]}
        value="lorem"
        onClose={handleClose}
        onNewTagClick={handleNewTagClick}
        onSave={handleSave}
        onTagChanged={handleTagChange}
      />,
    );

    const saveButton = getByTestId('dialog-save-button');
    fireEvent.click(saveButton);
    expect(handleSave).toHaveBeenCalledWith({
      comment: 'foo',
      id: '123',
      name: 'bar',
      value: 'lorem',
    });
  });

  test('should allow to close the dialog', () => {
    const handleClose = testing.fn();
    const handleNewTagClick = testing.fn();
    const handleSave = testing.fn();
    const handleTagChange = testing.fn();

    const {getByTestId} = render(
      <Dialog
        entitiesCount={3}
        tags={[{name: 'foo', id: '123'}]}
        onClose={handleClose}
        onNewTagClick={handleNewTagClick}
        onSave={handleSave}
        onTagChanged={handleTagChange}
      />,
    );

    const closeButton = getByTestId('dialog-close-button');

    fireEvent.click(closeButton);

    expect(handleClose).toHaveBeenCalled();
  });
});
