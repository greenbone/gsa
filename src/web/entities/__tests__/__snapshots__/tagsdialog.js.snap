// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsDialog dialog component tests should disable tag selection when no options available 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 {
  margin-left: -5px;
}

.c16>* {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c16>* {
  margin-left: 5px;
}

.c15 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c27 {
  cursor: pointer;
}

.c28 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c28 * {
  height: inherit;
  width: inherit;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 10% auto;
  border: 0;
  outline: 0;
  width: 650px;
  height: px;
}

.c0 {
  position: fixed;
  font-family: Trebuchet MS,Tahoma,Verdana,Arial,sans-serif;
  font-size: 1.1em;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin: 0;
  background: rgba(102, 102, 102, 0.5);
  z-index: 600;
  -webkit-transition: opacity 1s ease-in;
  transition: opacity 1s ease-in;
  width: 100%;
  height: 100%;
}

.c36 {
  width: 0;
  height: 0;
  cursor: nwse-resize;
  border-right: 20px solid transparent;
  border-bottom: 20px solid transparent;
  border-top: 20px solid #fff;
}

.c35 {
  position: absolute;
  bottom: 3px;
  right: 3px;
  width: 20px;
  height: 20px;
  background: repeating-linear-gradient(     -45deg,     transparent,     transparent 2px,     #000 2px,     #000 3px   );
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: inherit;
  padding: 0;
  background: #fff;
  box-shadow: 5px 5px 10px #7F7F7F;
  border-radius: 3px;
  border: 1px solid #7F7F7F;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
  font-size: 12px;
  font-family: Verdana,sans-serif;
  color: #074320;
  cursor: pointer;
  border-radius: 2px;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5 :hover {
  border: 1px solid #074320;
}

.c6 {
  height: 24px;
  width: 24px;
  line-height: 24px;
}

.c6 * {
  height: inherit;
  width: inherit;
}

.c32 {
  display: inline-block;
  padding: 0 15px;
  color: #4C4C4C;
  text-align: center;
  vertical-align: middle;
  font-size: 11px;
  font-weight: bold;
  line-height: 30px;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  background-color: #fff;
  border-radius: 2px;
  border: 1px solid #bfbfbf;
  cursor: pointer;
  overflow: visible;
  z-index: 1;
}

.c32:focus,
.c32:hover {
  border: 1px solid #4C4C4C;
}

.c32:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #11ab51;
  font-weight: bold;
  color: #fff;
}

.c32[disabled] {
  cursor: not-allowed;
  opacity: 0.65;
  box-shadow: none;
}

.c32 img {
  height: 32px;
  width: 32px;
  margin-top: 5px 10px 5px -10px;
  vertical-align: middle;
}

.c32:link {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #4C4C4C;
}

.c33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c34 {
  border: 1px solid #7F7F7F;
  color: #fff;
  background: #11ab51;
}

.c34 :hover {
  color: #074320;
  background: #A1DDBA;
}

.c30 {
  border-width: 1px 0 0 0;
  border-style: solid;
  border-color: #e5e5e5;
  margin-top: 15px;
  padding: 10px 20px 10px 20px;
}

.c31 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.c4 {
  padding: 5px 5px 5px 10px;
  margin-bottom: 15px;
  border-radius: 2px 2px 0 0;
  border-bottom: 1px solid #7F7F7F;
  color: #fff;
  font-weight: bold;
  background: #11ab51;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: -webkit-grab;
  cursor: grab;
}

.c9 {
  overflow: auto;
  padding: 0 15px;
  width: 100%;
  height: 100%;
  max-height: 400px;
}

.c8 {
  overflow: hidden;
  height: 100%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  padding-bottom: 10px;
}

.c12 {
  display: inline-block;
  max-width: 100%;
  font-weight: bold;
  text-align: right;
  padding-left: 10px;
  padding-right: 10px;
  width: 33.33333333%;
  margin-left: 0;
}

.c14 {
  width: 66.66666667%;
  padding-left: 10px;
  padding-right: 10px;
}

.c23 {
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline: none;
  margin: 1px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c24 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c24 * {
  height: inherit;
  width: inherit;
}

.c19 {
  border: 1px solid #bfbfbf;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 1px 5px;
  background-color: #fff;
  color: #000;
  font-weight: normal;
  background-color: #f3f3f3;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 230px;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  word-break: keep-all;
  white-space: nowrap;
  overflow: hidden;
  cursor: default;
}

.c25 {
  color: #c12c30;
  font-weight: bold;
  font-size: 19px;
  padding-bottom: 1px;
  padding-left: 4px;
  display: none;
}

.c21 {
  cursor: default;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media print {
  .c26 {
    display: none;
  }
}

<body>
  <div
    id="portals"
  >
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
          height="px"
          role="dialog"
          tabindex="1"
          width="650px"
        >
          <div
            class="c2"
          >
            <div
              class="c3 c4"
              data-testid="dialog-title-bar"
            >
              <span>
                Add Tag
              </span>
              <div
                class="c5 c6"
                data-testid="dialog-title-close-button"
                title="Close"
              >
                ×
              </div>
            </div>
            <div
              class="c7 c8"
            >
              <div
                class="c9"
                data-testid="save-dialog-content"
              >
                <div
                  class="c10"
                >
                  <div
                    class="c11"
                  >
                    <label
                      class="c12"
                      data-testid="formgroup-title"
                    >
                      Choose Tag
                    </label>
                    <div
                      class="c13 c14"
                      data-testid="formgroup-content"
                    >
                      <div
                        class="c13 c15"
                      >
                        <div
                          class="c13 c16"
                        >
                          <div
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            aria-labelledby="downshift-1-label"
                            class="c17"
                            role="combobox"
                          >
                            <div
                              class="c18"
                              width="230px"
                            >
                              <div
                                aria-haspopup="true"
                                aria-label="open menu"
                                class="c19"
                                data-toggle="true"
                                disabled=""
                                role="button"
                                type="button"
                              >
                                <div
                                  class="c20 c21"
                                  data-testid="select-selected-value"
                                  disabled=""
                                />
                                <div
                                  class="c22"
                                >
                                  <span
                                    class="c23 c24"
                                    data-testid="select-open-button"
                                  >
                                    ▼
                                  </span>
                                </div>
                              </div>
                            </div>
                            <div
                              class="c25"
                              data-testid="error-marker"
                            >
                              ×
                            </div>
                          </div>
                          <span
                            class="c26 c27 c28"
                            data-testid="svg-icon"
                            title="Create a new Tag"
                          >
                            <svg
                              title="Create a new Tag"
                            >
                              new.svg
                            </svg>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="c11"
                  >
                    <label
                      class="c12"
                      data-testid="formgroup-title"
                    >
                      Value
                    </label>
                    <div
                      class="c13 c14"
                      data-testid="formgroup-content"
                    />
                  </div>
                  <div
                    class="c11"
                  >
                    <label
                      class="c12"
                      data-testid="formgroup-title"
                    >
                      Comment
                    </label>
                    <div
                      class="c13 c14"
                      data-testid="formgroup-content"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="c29 c30 c31"
            >
              <button
                class="c32 c33 c34"
                data-testid="dialog-close-button"
                title="Cancel"
              >
                Cancel
              </button>
              <button
                class="c32 c33 c34"
                data-testid="dialog-save-button"
                title="Add Tag"
              >
                Add Tag
              </button>
            </div>
          </div>
          <div
            class="c35"
          >
            <div
              class="c36"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</body>
`;

exports[`TagsDialog dialog component tests should render dialog 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 {
  margin-left: -5px;
}

.c16>* {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c16>* {
  margin-left: 5px;
}

.c15 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c27 {
  cursor: pointer;
}

.c28 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c28 * {
  height: inherit;
  width: inherit;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 10% auto;
  border: 0;
  outline: 0;
  width: 650px;
  height: px;
}

.c0 {
  position: fixed;
  font-family: Trebuchet MS,Tahoma,Verdana,Arial,sans-serif;
  font-size: 1.1em;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin: 0;
  background: rgba(102, 102, 102, 0.5);
  z-index: 600;
  -webkit-transition: opacity 1s ease-in;
  transition: opacity 1s ease-in;
  width: 100%;
  height: 100%;
}

.c36 {
  width: 0;
  height: 0;
  cursor: nwse-resize;
  border-right: 20px solid transparent;
  border-bottom: 20px solid transparent;
  border-top: 20px solid #fff;
}

.c35 {
  position: absolute;
  bottom: 3px;
  right: 3px;
  width: 20px;
  height: 20px;
  background: repeating-linear-gradient(     -45deg,     transparent,     transparent 2px,     #000 2px,     #000 3px   );
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: inherit;
  padding: 0;
  background: #fff;
  box-shadow: 5px 5px 10px #7F7F7F;
  border-radius: 3px;
  border: 1px solid #7F7F7F;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
  font-size: 12px;
  font-family: Verdana,sans-serif;
  color: #074320;
  cursor: pointer;
  border-radius: 2px;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5 :hover {
  border: 1px solid #074320;
}

.c6 {
  height: 24px;
  width: 24px;
  line-height: 24px;
}

.c6 * {
  height: inherit;
  width: inherit;
}

.c32 {
  display: inline-block;
  padding: 0 15px;
  color: #4C4C4C;
  text-align: center;
  vertical-align: middle;
  font-size: 11px;
  font-weight: bold;
  line-height: 30px;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  background-color: #fff;
  border-radius: 2px;
  border: 1px solid #bfbfbf;
  cursor: pointer;
  overflow: visible;
  z-index: 1;
}

.c32:focus,
.c32:hover {
  border: 1px solid #4C4C4C;
}

.c32:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #11ab51;
  font-weight: bold;
  color: #fff;
}

.c32[disabled] {
  cursor: not-allowed;
  opacity: 0.65;
  box-shadow: none;
}

.c32 img {
  height: 32px;
  width: 32px;
  margin-top: 5px 10px 5px -10px;
  vertical-align: middle;
}

.c32:link {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #4C4C4C;
}

.c33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c34 {
  border: 1px solid #7F7F7F;
  color: #fff;
  background: #11ab51;
}

.c34 :hover {
  color: #074320;
  background: #A1DDBA;
}

.c30 {
  border-width: 1px 0 0 0;
  border-style: solid;
  border-color: #e5e5e5;
  margin-top: 15px;
  padding: 10px 20px 10px 20px;
}

.c31 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.c4 {
  padding: 5px 5px 5px 10px;
  margin-bottom: 15px;
  border-radius: 2px 2px 0 0;
  border-bottom: 1px solid #7F7F7F;
  color: #fff;
  font-weight: bold;
  background: #11ab51;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: -webkit-grab;
  cursor: grab;
}

.c9 {
  overflow: auto;
  padding: 0 15px;
  width: 100%;
  height: 100%;
  max-height: 400px;
}

.c8 {
  overflow: hidden;
  height: 100%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  padding-bottom: 10px;
}

.c12 {
  display: inline-block;
  max-width: 100%;
  font-weight: bold;
  text-align: right;
  padding-left: 10px;
  padding-right: 10px;
  width: 33.33333333%;
  margin-left: 0;
}

.c14 {
  width: 66.66666667%;
  padding-left: 10px;
  padding-right: 10px;
}

.c23 {
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline: none;
  margin: 1px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c24 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c24 * {
  height: inherit;
  width: inherit;
}

.c19 {
  border: 1px solid #bfbfbf;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 1px 5px;
  background-color: #fff;
  color: #000;
  font-weight: normal;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 230px;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  word-break: keep-all;
  white-space: nowrap;
  overflow: hidden;
  cursor: pointer;
}

.c25 {
  color: #c12c30;
  font-weight: bold;
  font-size: 19px;
  padding-bottom: 1px;
  padding-left: 4px;
  display: none;
}

.c21 {
  cursor: default;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media print {
  .c26 {
    display: none;
  }
}

<body>
  <div
    id="portals"
  >
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
          height="px"
          role="dialog"
          tabindex="1"
          width="650px"
        >
          <div
            class="c2"
          >
            <div
              class="c3 c4"
              data-testid="dialog-title-bar"
            >
              <span>
                Add Tag
              </span>
              <div
                class="c5 c6"
                data-testid="dialog-title-close-button"
                title="Close"
              >
                ×
              </div>
            </div>
            <div
              class="c7 c8"
            >
              <div
                class="c9"
                data-testid="save-dialog-content"
              >
                <div
                  class="c10"
                >
                  <div
                    class="c11"
                  >
                    <label
                      class="c12"
                      data-testid="formgroup-title"
                    >
                      Choose Tag
                    </label>
                    <div
                      class="c13 c14"
                      data-testid="formgroup-content"
                    >
                      <div
                        class="c13 c15"
                      >
                        <div
                          class="c13 c16"
                        >
                          <div
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            aria-labelledby="downshift-0-label"
                            class="c17"
                            role="combobox"
                          >
                            <div
                              class="c18"
                              width="230px"
                            >
                              <div
                                aria-haspopup="true"
                                aria-label="open menu"
                                class="c19"
                                data-toggle="true"
                                role="button"
                                type="button"
                              >
                                <div
                                  class="c20 c21"
                                  data-testid="select-selected-value"
                                />
                                <div
                                  class="c22"
                                >
                                  <span
                                    class="c23 c24"
                                    data-testid="select-open-button"
                                  >
                                    ▼
                                  </span>
                                </div>
                              </div>
                            </div>
                            <div
                              class="c25"
                              data-testid="error-marker"
                            >
                              ×
                            </div>
                          </div>
                          <span
                            class="c26 c27 c28"
                            data-testid="svg-icon"
                            title="Create a new Tag"
                          >
                            <svg
                              title="Create a new Tag"
                            >
                              new.svg
                            </svg>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="c11"
                  >
                    <label
                      class="c12"
                      data-testid="formgroup-title"
                    >
                      Value
                    </label>
                    <div
                      class="c13 c14"
                      data-testid="formgroup-content"
                    />
                  </div>
                  <div
                    class="c11"
                  >
                    <label
                      class="c12"
                      data-testid="formgroup-title"
                    >
                      Comment
                    </label>
                    <div
                      class="c13 c14"
                      data-testid="formgroup-content"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="c29 c30 c31"
            >
              <button
                class="c32 c33 c34"
                data-testid="dialog-close-button"
                title="Cancel"
              >
                Cancel
              </button>
              <button
                class="c32 c33 c34"
                data-testid="dialog-save-button"
                title="Add Tag"
              >
                Add Tag
              </button>
            </div>
          </div>
          <div
            class="c35"
          >
            <div
              class="c36"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</body>
`;
